### PR TITLE
develop_-_moving_images_to_database

### PR DESCRIPTION
**Pull request by syphzur**

### CODE REVIEW

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private ImageService imageService;
+
+    public ImageController(final ImageService imageService) {
+        this.imageService = imageService;
+    }
+
+    @GetMapping(path = "/{code}", produces = MediaType.IMAGE_PNG_VALUE)
+    @PreAuthorize("hasPermission('image', 'single' )")
+    public ResponseEntity<ByteArrayResource> getSingle(@PathVariable("code") final String code) {
+        final ImageModelDto img = imageService.getSingleObject(code);
```
Comment (latusikl) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java:33]: Does it make sense to create Dto? Maybe passing image directly would be better if dto is not returned to user?

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        final ByteArrayResource resource = new ByteArrayResource(img.getBytes());
+        return new ResponseEntity<>(resource, HttpStatus.OK);
+    }
+
+    @PostMapping
+    @PreAuthorize("hasPermission('image', 'post' )")
+    public ResponseEntity postImage(@Image(service = ImageService.class) final MultipartFile file) throws IOException {
+        return new ResponseEntity(imageService.save(List.of(
+                new ImageModelDto(null, file.getOriginalFilename(), file.getBytes()))));
```
Comment (latusikl) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java:42]: Same here

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/entities/Image.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Entity(name = "images")
+@Table(name = "images")
+@Getter
+@Setter
+@NoArgsConstructor
+public class Image extends AbstractEntityModel {
+
+    public Image(final String name, final byte[] bytes) {
+        this.name = name;
+        this.bytes = bytes;
```
Comment (latusikl) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/entities/Image.java:22]: You can try usage of @AllArgsConstructor. See [documentatnion](https://projectlombok.org/api/lombok/AllArgsConstructor.html)

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/dto/ImageModelDto.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import pl.polsl.photoplus.annotations.Patchable;
+import pl.polsl.photoplus.annotations.validators.Unique;
+import pl.polsl.photoplus.services.controllers.ImageService;
+
+import javax.persistence.Lob;
+import javax.validation.constraints.NotBlank;
+
+@Getter
+@Setter
+public class ImageModelDto extends AbstractModelDto<ImageModelDto> {
```
Comment (latusikl) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/dto/ImageModelDto.java:14]: Is this needed in picture controller we do not receive this pojo?

**Review by latusikl — APPROVED** [review]

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/dto/ImageModelDto.java (hunk 4 above)
Comment (syphzur) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/model/dto/ImageModelDto.java:14]: it's needed to use generic service

**Pull request merged**